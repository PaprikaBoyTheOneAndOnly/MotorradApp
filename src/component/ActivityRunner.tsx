import React, {Component} from "react";
import {ActivityIndicator, StyleSheet, Text, View} from "react-native";
import {globalStyles} from "../data.module";

interface IProps {
    text: string
}
export default class ActivityRunner extends Component<IProps,{}> {
    render() {
       return (
        <View style={styles.container}>
            <ActivityIndicator size="large" color="#000000"/>
            <Text>{`\n${this.props.text}`}</Text>
        </View>)
    }
}

const styles = StyleSheet.create(
// @ts-ignore
    {
        ...globalStyles,
    }
);
