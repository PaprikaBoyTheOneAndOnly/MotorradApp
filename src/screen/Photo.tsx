import React, {Component} from 'react';
import {Dimensions, Image as NativeImage, StyleSheet, TouchableOpacity, View, Text, Alert} from 'react-native';
import {globalStyles, IPhoto, stackConfig} from '../data.module';
import {StackNavigationProp} from '@react-navigation/stack';
import {Image} from 'react-native-elements';
import ActivityRunner from '../component/ActivityRunner';
import * as FirebaseService from '../service/FirebaseService';
import {extractNameOfPhoto} from '../service/FirebaseService';

interface IProps {
    navigation: StackNavigationProp<any>;
}

interface IState {
    photo: IPhoto,
    width: number,
    height: number,
}

export default class Photo extends Component<IProps, IState> {
    static navigationOptions = ({navigation}) => ({
        ...stackConfig,
        headerTitle: 'Photo',
        headerRight: () => {
            const onPressDelete = () =>
                FirebaseService.deletePhoto(navigation.state.params.routeName,
                    extractNameOfPhoto(navigation.state.params.photo.url))
                    .then(() => navigation.goBack())
                    .catch(() => Alert.alert('Error', 'Could not delete photo!' +
                        '\nPlease try again later.'));

            return (
                <TouchableOpacity style={{marginRight: 20}} onPress={onPressDelete}>
                    <Text style={{color: 'red'}}>Delete</Text>
                </TouchableOpacity>
            );
        }
    });

    constructor(props) {
        super(props);
        this.state = {
            photo: this.props.navigation.getParam('photo'),
            width: null,
            height: null,
        };
    }

    componentDidMount() {
        NativeImage.getSize(this.state.photo.url, (width, height) =>
            this.setState({width, height}), null);
    }

    render() {
        if (this.state.height && this.state.width) {
            const width = Dimensions.get('window').width;
            const ratio = width / this.state.width;
            const style = {
                width: width,
                height: this.state.height * ratio,
            };

            return (
                <View style={{...styles.container, backgroundColor: 'black'}}>
                    <Image source={{uri: this.state.photo.url}} style={style}/>
                </View>
            );
        }
        return <ActivityRunner text={'Waiting for image'}/>
    }
}

const styles = StyleSheet.create(
// @ts-ignore
    {
        ...globalStyles,
    }
);
