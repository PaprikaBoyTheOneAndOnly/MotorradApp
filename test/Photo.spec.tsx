import {shallow} from 'enzyme';
import Photo from "../src/screen/Photo";
import {IPhoto} from "../src/data.module";
import {StackNavigationProp} from "@react-navigation/stack";
import React from "react";
import ActivityRunner from "../src/component/ActivityRunner";

const mockNavigation: StackNavigationProp<any> = {
    getParam: (any: string) => {
        const photo: IPhoto = {
            url: 'photoUrl.com',
            coordinate: {
                longitude: 1,
                latitude: 1,
            }
        };
        return photo
    }
};

describe('Photo', () => {
    let component;

    beforeEach(() => {
        component = shallow(<Photo navigation={mockNavigation}/>);
        component.setState({width: 100, height: 100});
    });

    it('should match snapshot', () => {
        expect(component.not.contains(<ActivityRunner text={'Waiting for image'}/>));
        expect(component).toMatchSnapshot();
    });


    it('should match snapshot when state isn\'t updated so far', () => {
        const component = shallow(<Photo navigation={mockNavigation}/>);
        expect(component.contains(<ActivityRunner text={'Waiting for image'}/>));
       
        expect(component).toMatchSnapshot();
    });

    xit('should calculate the height by the width',() => {
        // how can i mock Dimensions
        jest.mock('Dimensions', () => {
            // const Dim = Dimensions as jest.Mocked<typeof Dimensions>;
            // Dim.get = jest.fn(() => ({width: 100, height: 100}));
            // return Dimensions;
        });
    });

    xit('should delete photo', () => {
        const mockDeletePhoto = jest.fn(() => Promise.resolve());
        jest.mock('../src/service/FirebaseService', () => ({
            deletePhoto: mockDeletePhoto,
        }));

        //how can i access the onPress function on the TouchableOpacity in the navigationOption?
        console.log(component.find('TouchableOpacity'));
        expect(mockDeletePhoto).toHaveBeenCalled();
        mockDeletePhoto.mockClear();
    });
});