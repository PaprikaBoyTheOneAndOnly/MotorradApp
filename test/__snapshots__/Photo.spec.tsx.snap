// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo should match snapshot 1`] = `ShallowWrapper {}`;

exports[`Photo should match snapshot when state isn't updated so far 1`] = `ShallowWrapper {}`;
